### PR TITLE
[NG][WIP] Build refactoring

### PR DESCRIPTION
**Pull request by GuillaumeNury**
⚠️ **DO NOT MERGE** ⚠️ 

**TLDR:** This branch is a prototype, it solves some problems but create another

* Add [ng-packagr](https://github.com/dherges/ng-packagr) to bundle the app

To do so, I had to move all `ng` code in `ng/lib` in order to have a "clean" `package.json` at the root level (without all dependancies)

With this PR, you do not have to import `@lucca-front/ng/src` anymore; but it add one additional problem : as this package is not build with consumer app anymore, the style will be compiled during the bundling, with @lucca-front theme (not the theme of the consumer).
Another problem is that all consumer apps have to add formly in their dependancies.

Here is a list of the perfect consuming experience:

- [x] `import { XXX } from '@lucca-front/ng'`
- [ ] Formy in a separate package (`@lucca-front/ng/formly` or `@lucca-front/ng-formly`)
- [ ] Overridable theme

Another things to do:

- [ ] Update the publish process
- [ ] Remove old build process (part of gulp)
- [ ] Test all other npm scripts

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
Travis automatic deployment: https://277-pr-lucca-front-LuccaSA.surge.sh
